lib/cmds/blockchain/geth_commands.js

```diff
@@ -12,83 +12,83 @@ class GethCommands {
 
   commonOptions() {
     let config = this.config;
-    let cmd = "";
+    let cmd = [];
 
-    cmd += this.determineNetworkType(config);
+    cmd.push(this.determineNetworkType(config));
 
     if (config.datadir) {
-      cmd += "--datadir=\"" + config.datadir + "\" ";
+      cmd.push(`--datadir=${config.datadir}`);
     }
 
     if (config.light) {
-      cmd += "--light ";
+      cmd.push("--light");
     }
 
     if (config.fast) {
-      cmd += "--fast ";
+      cmd.push("--fast");
     }
 
     if (config.account && config.account.password) {
-      cmd += "--password " + config.account.password + " ";
+      cmd.push(`--password=${config.account.password}`);
     }
 
     if (Number.isInteger(config.verbosity) && config.verbosity >=0 && config.verbosity <= 5) {
-      cmd += "--verbosity " + config.verbosity + " ";
+      cmd.push("--verbosity=" + config.verbosity);
     }
 
     return cmd;
   }
 
-  determineVersion() {
+  determineVersionCommand() {
     return this.geth_bin + " version";
   }
 
   determineNetworkType(config) {
-    let cmd = "";
+    let cmd;
     if (config.networkType === 'testnet') {
-      cmd += "--testnet ";
+      cmd = "--testnet ";
     } else if (config.networkType === 'olympic') {
-      cmd += "--olympic ";
+      cmd = "--olympic ";
     } else if (config.networkType === 'custom') {
-      cmd += "--networkid " + config.networkId + " ";
+      cmd = "--networkid=" + config.networkId;
     }
     return cmd;
   }
 
   initGenesisCommmand() {
     let config = this.config;
-    let cmd = this.geth_bin + " " + this.commonOptions();
+    let cmd = this.geth_bin + " " + this.commonOptions().join(' ');
 
     if (config.genesisBlock) {
-      cmd += "init \"" + config.genesisBlock + "\" ";
+      cmd += " init \"" + config.genesisBlock + "\" ";
```

Comment:
There a reason for the space here?

Reply:
Since I changed from strings of arguments to arrays, I lost the final space, so I needed to put the space there, otherwise it was something like `--arg 100init genesisblock`

Reply:
got it.